### PR TITLE
Document on collaborative workflows

### PR DESCRIPTION
**Pull request by mih**
Closes https://github.com/psychoinformatics-de/datalad-debian/issues/35

### CODE REVIEW

File: docs/source/walkthrough_collab.rst [hunk 1]
```diff
@@ -0,0 +1,499 @@
...
+
+- **APT-repository dataset**: DataLad dataset that tracks the content of an APT
+  repository that would typically be exposed via a web-server (i.e., crucially the
+  `dist/`, and `pool/` directories). For technical or legal reasons (number of
+  files, split free and non-free components, etc.) this dataset may optionally have
+  additional subdatasets.
+
+- **archive dataset**: DataLad dataset that tracks an *APT-repository dataset*,
+  and any number of *distribution datasets* used to populate the APT-repository
+  (presently with the `reprepro` tool).
```
Comment (adswa) [inline docs/source/walkthrough_collab.rst:43]:
```suggestion
  (presently with the `reprepro <https://salsa.debian.org/brlink/reprepro>`_ tool).
```

File: docs/source/walkthrough_collab.rst [hunk 2]
```diff
@@ -0,0 +1,499 @@
...
+datasets. Individual roles people can have are associated with different sets
+of required access permissions.
+
+- **package maintainer**: Updates a *package dataset* with new (backported)
+  versions of a source package. Needs read/write access to a (particular)
+  *package dataset*.
+
+- **distribution maintainer**: Reviews and accepts/rejects new or updated
+  *package datasets* (updated by a *package maintainer* or a *package
+  builders*) by incorporating them into a *distribution dataset*. Needs read
```
Comment (adswa) [inline docs/source/walkthrough_collab.rst:60]:
```suggestion
  builder*) by incorporating them into a *distribution dataset*. Needs read
```

File: docs/source/walkthrough_collab.rst [hunk 3]
```diff
@@ -0,0 +1,499 @@
...
+performing updates. For small efforts, even the simplistic approach of using a
+monolithic directory (as shown in the :ref:`initial walk-through
+<chap_walkthrough>`) may be good enough. However, larger operations involving
+multiple people and different levels of trust benefit from a stricter
+compartmentalization.
+
+- **APT-repository dataset checkout**: This location is typically exposed via a
+  web-server and the APT client facing end of the system. In addition to being
+  used by APT for its normal operation, it can also be (git/datalad) cloned
+  from (including historic version), and is capable of hosting and providing
```
Comment (adswa) [inline docs/source/walkthrough_collab.rst:99]:
```suggestion
  from (including historic versions), and is capable of hosting and providing
```

File: docs/source/walkthrough_collab.rst [hunk 4]
```diff
@@ -0,0 +1,499 @@
...
+
+- **Hosting of distribution datasets**: These need to have read access for all
+  maintainer roles, and write access for *distribution maintainers* only. Larger
+  efforts might benefit from a proper Git hosting solution in order to be able
+  to process incoming package update requests via a pull-request or issue
+  tracking system.
+
+- **Hosting of package datasets**: *Package maintainers* need to deposit source
+  and binary package files in a way that enables *package builders* and
+  *archive maintainers* to retrieve them. Any datalad-supported sibling type can
```
Comment (adswa) [inline docs/source/walkthrough_collab.rst:119]:
```suggestion
  *archive maintainers* to retrieve them. Any DataLad-supported sibling type can
```

File: docs/source/walkthrough_collab.rst [hunk 5]
```diff
@@ -0,0 +1,499 @@
...
+Create an archive dataset
+-------------------------
+
+An *archive dataset* has different hosting demands. The ``reprepro`` tool
+essentially requires the full work tree to be present at all times. For large
+APT-archives even a clone may take considerable time. Hence we are create the
+archive dataset in the location where is would/could live semi-persistently.
+
+We add our *distribution dataset* from the collaboration-focused dataset store
+(GitLab placeholder). The need not live on the same machine. Any source URL
```
Comment (adswa) [inline docs/source/walkthrough_collab.rst:230]:
```suggestion
(GitLab placeholder). They need not live on the same machine. Any source URL
```

File: docs/source/walkthrough_collab.rst [hunk 6]
```diff
@@ -0,0 +1,499 @@
...
+Here we clone via SSH to indicate that this could be performed anywhere.
+
+.. code-block:: bash
+
+   datalad clone --reckless ephemeral ria+ssh://localhost/tmp/wt/gitlab#~pkg-demo pkg
+
+A key task of updating a *package dataset* is adding a new source package version.
+This can involve arbitrary procedures. Here we simply download a ready-made
+source package from Debian. Alternatively, a source package could be generated
+via `git-buildpackage` from a linked packaging repo, or something equivalent.
```
Comment (adswa) [inline docs/source/walkthrough_collab.rst:307]:
```suggestion
via `git-buildpackage <https://honk.sigxcpu.org/projects/git-buildpackage/manual-html/>`_ from a linked packaging repo, or something equivalent.
```

**Review by adswa — COMMENTED** [review]
quick fixes for a few typos, and links to tools non-debian people like me might be unfamiliar with

**Comment by mih** [comment]
OK, let's merge and improve from here.

**Pull request merged**